pkg/providers/cloudstack/validator.go

```diff
@@ -61,6 +62,14 @@ func (v *Validator) ValidateCloudStackDatacenterConfig(ctx context.Context, data
 	if err != nil {
 		return fmt.Errorf("error while checking management api endpoint: %v", err)
 	}
+	execConfig, err := decoder.ParseCloudStackSecret()
+	if err != nil {
+		return fmt.Errorf("parsing cloudstack secret: %v", err)
+	}
+	if execConfig.ManagementUrl != datacenterConfig.Spec.ManagementApiEndpoint {
+		return fmt.Errorf("cloudstack secret management url (%s) differs from cluster spec management url (%s)",
```

Comment:
This is probably a great error message. Im wondering if it makes sense to also have a validation if the ManagementApiEndpoint provided on cluster spec to be non-empty, and throw an appropriate message at that point?

Reply:
Sure, I can add another check on the provided spec's managementApiEndpoint to make sure it's nonempty

Reply:
actually the managementApiEndpoint is a required field per https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go#L36. Does that error on empty string inputs?